baseline/tf/tagger/model.py

```diff
@@ -344,43 +315,132 @@ def create(cls, embeddings, labels, **kwargs):
             model.y = kwargs.get('y', tf.compat.v1.placeholder(tf.int32, [None, None], name="y"))
             model.sess = kwargs.get('sess', create_session())
 
-        model._record_state(**kwargs)
+        model._record_state(embeddings, **kwargs)
         model.labels = labels
-        nc = len(labels)
-
         # This only exists to make exporting easier
         model.pdrop_value = kwargs.get('dropout', 0.5)
         model.dropin_value = kwargs.get('dropin', {})
 
         model.labels = labels
         model.span_type = kwargs.get('span_type')
 
-        embed_model = model.embed(**kwargs)
-        transduce_model = model.encode(**kwargs)
-        decode_model = model.decode(**kwargs)
-
-        model.impl = TagSequenceModel(nc, embed_model, transduce_model, decode_model)
+        model.create_layers(embeddings, **kwargs)
         if not tf.executing_eagerly():
-            model.probs = model.impl.transduce(inputs)
-            model.best = model.impl.decode(model.probs, model.lengths)
+            model.best = model(inputs)
         return model
 
+    def create_loss(self):
+        """Create the loss function and return it
 
-@register_model(task='tagger', name='default')
-class RNNTaggerModel(TaggerModelBase):
+        :return: The loss function
+        """
 
-    @property
-    def vdrop(self):
-        return self._vdrop
 
-    @vdrop.setter
-    def vdrop(self, value):
-        self._vdrop = value
+class TransducerTaggerModel(TaggerModelBase):
 
     def __init__(self):
         super().__init__()
 
-    def encode(self, **kwargs):
+    def create_layers(self, embeddings: Dict[str, TensorDef], **kwargs):
+        self.embeddings = self.init_embed(embeddings, **kwargs)
+        self.encoder = self.init_encode(**kwargs)
+        self.proj_layer = tf.keras.layers.Dense(len(self.labels))
```

Comment:
I know there was some talk of making this projection layer optional for things like the LAN tagger where the encoder output is already the same size as the output but it seems like we can't check for that at build time in TF. Does it make sense to create a `ProjectionLayer` object in `eight_mile` where if the input and output size are the same it does nothing and if they are different it does uses a Dense layer to do the projection?

Reply:
IMO what makes sense is add an `output_dim` to the BiLSTM encoder in TF like we have in PyTorch.  The proj_layer check could then be a test against the `output_dim` (if they are the same, create, if not dont).  In some case it might still be desirable to explicitly tweak, so another simple change would be to have a `self.init_proj()` step

Reply:
I added an `init_proj` layer which you can override if you dont like the linear projection applied.  For instance, you could use a skip conn or you could just use an `PassThru/Identity` in the case that you already have logits coming out of the encoder layer.  I also updated the docs to describe this as "step 3" of the model